enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -0,0 +1,972 @@
+      >   - Being in this place in decision tree means that either restore failed or system was unhealthy.
+      >     Restoring again seems to best bet.
+      >
+      > - If this is manual reboot, then system was healthy,
+      >   then manually rebooted, backed data up and end up unhealthy,
+      >   again rebooted resulting in being in this place in decision tree.
+      >   - Admin should address problems before rebooting the system
+      >     (and retrigger greenboot first, to refresh system's health)
+      >   - Restoring data seems okay - going back to last healthy state.
+
+   1. If backup for current commit does not exist
+      > There's no proof of the commit being ever healthy
+
+      1. If "history of commits" does not know about commit before current one: **delete data and [start cluster](#starting-the-cluster)**
+         > First commit with MicroShift running on the system, system is consistently unhealthy.
+
+      1. If "history of commits" has entry about commit before current one
+         > MicroShift was already running on the device.
+         > But it doesn't mean system will rollback to that "commit before current one"
+
+         1. "Commit before current one" is the same as rollback according to ostree
+            1. If version metadata matches "commit before current one": **backup data and proceed to [data migration](#data-migration-1)**
+               > Previous boot of current commit might have been unhealthy because it failed to make a backup.
+
+            1. Otherwise: **restore backup of "commit before current one"**
+               > Give chance to migrate data and start cluster again.
+               > Assumption that admin upgraded from healthy system is important here.
+
+         1. "Commit before current one" is not the rollback: **delete data and [start cluster](#starting-the-cluster)**
+            > Means that rollback does not feature MicroShift.
+            > This is "retry boot" of FIDO scenario.
+
+##### Data migration
+
+1. Compare version persisted in metadata with MicroShift's binary
+   - Binary's `Y` is smaller: **abort and block cluster start up**
+   - Binary's `Y` is bigger by more than `1`: **abort and block cluster start up**
+   - Version in metadata is present in "list of prohibited migrations:
+     **abort and block cluster start up**
+   - `Y`s are the same: **skip to cluster start up**
+1. Perform data migration
+
+##### Starting the cluster
+1. If metadata exists and it doesn't match version of the binary: **abort**
+   > Extra check to make sure that migration was performed
+1. Create data dir if necessary
+1. Create or update metadata (version and "history of commits")
+1. Continue regular flow
+
+##### Health check
+1. Assess health of MicroShift and persist the result to "history of commits"
+
+##### MicroShift's green and red scripts
+1. Write system's health to "history of commits"
+
+#### Manual interventions
```

Comment:
An introduction could help this section. You could explain that the scenarios are descriptions of the steps applied in specific situations and are derived from the decision tree presented earlier, for example, and are not distinct from it. That would also apply to other sections at this heading level, like "First ostree commit" and "Second ostree commit".